### PR TITLE
Use original channel data on the night side for NIR emissive

### PR DESCRIPTION
**Pull request by pnuu**
In Satpy any composite that uses `nir_emissive` enhancement has the whole night side masked. This happens because the emissive part was calculated by subtracting the reflective part and subtracting this from the full radiance, and the reflective part is masked where the Sun zenith angle is greater than 88 degrees.

This PR uses the original brightness temperature data beyond this 88 degree SZA limit.

 - [ ] Tests added 
 - [ ] Tests passed: Passes ``pytest pyspectral`` 
 - [x] Passes ``flake8 pyspectral``

### CODE REVIEW

File: pyspectral/near_infrared_reflectance.py [hunk 1]
```diff
@@ -252,7 +254,9 @@ def reflectance_from_tbs(self, sun_zenith, tb_near_ir, tb_thermal, **kwargs):
         self._rad3x_t11 = self.tb2radiance(tb_therm, lut=lut)['radiance']
         thermal_emiss_one = self._rad3x_t11 * self.rsr_integral
 
-        l_nir = self.tb2radiance(tb_nir, lut=lut)['radiance'] * self.rsr_integral
+        l_nir = self.tb2radiance(tb_nir, lut=lut)['radiance']
+        self._rad3x = l_nir.copy()
+        l_nir *= self.rsr_integral
```
Comment (mraspaud) [inline pyspectral/near_infrared_reflectance.py]: Could you explain the motivation here ?

**Comment by mraspaud** [comment]
Looks good so far! Just waiting for tests now ;)

File: pyspectral/near_infrared_reflectance.py (hunk 1 above)
Comment (pnuu) [reply pyspectral/near_infrared_reflectance.py]: The night side needs to be set with the unmodified radiances. The multiplication with the RSR integral breaks changes the data to be used in some mystical way I didn't go through.

Comment (mraspaud) [reply pyspectral/near_infrared_reflectance.py]: Understood, thanks.

**Comment by pnuu** [comment]
I think this will need much more in-depth understanding what is done to retrieve the emissive part than I have. Or have time to think. Consider this as the first iteration of "the night side shouldn't be masked" :-)

The day-time emissive part is - if I understood correctly the quick browse of the code - is retrived using ~10.8 um data somehow. The way I have it now assumed the day-time data would approach the night side, but there's a rather big step close to terminator.

**Comment by mraspaud** [comment]
A smooth transition would be nicer to look at, but less correct, right?

**Comment by pnuu** [comment]
Weeeeell, if we just consider what the emissive part retrieval is doing, it *should* match the part that has zero reflective part, i.e. the night side. There will of course be e.g. illuminated clouds and stuff there.. But maybe, as the emissive retrieval stops at 88 def SZA, we could to linear interpolation for a transitional zone? Or would that be too computationally expensive?

**Comment by adybbroe** [comment]
Thanks Panu, I agree this is fine for now, good to have this sorted out, not masking the nightside. About the transition, it is more challenging, and I am not sure this is something we should provide in Pyspectral.

**Pull request merged**